Scrapers/VenueExplorer.py

```diff
@@ -1 +1,130 @@
-# TODO: Write scraping script here
\ No newline at end of file
+from bs4 import BeautifulSoup
+from requests import get
+import VenueClass
+import json
+import os
+
+allVenues = []  # Array to store all Venues of type VenueClass
+
+def extractVenue(venue):
+
+    # extract title
+    title_html = venue.find('div', class_='event-name')
+    title = ''
+    if title_html:
+        title = title_html.text
+
+    # extract tags
+    venue_tags = venue.find('b', class_='t-holder').text.split(",")
+    tags = []
+    for venue_tag in venue_tags:
+        tags.append(venue_tag.strip())
+
+    # extract link into the venue
+    venue_link = venue.find_all('a')[1]['href']
+    response = get(venue_link)
+    indiv_venue = BeautifulSoup(response.text, 'html.parser')
+
+    # extract image links
+    venue_images = indiv_venue.find_all('a', class_='example-image-link')
+    images = []
+    for venue_image in venue_images:
+        image = 'venuexplorer.com.sg' + str(venue_image['href'])
+        images.append(image)
+
+    # extract starting price - same for all rooms within this venue
+    price_html = indiv_venue.find_all('div', class_='col-lg-3 col-md-3 col-sm-3 text-center')[1]
+    price = ''
+    if price_html:
+        price = price_html.text.strip()
+
+    # extract general capacity
+    capacity_html = indiv_venue.find_all('div', class_='col-lg-3 col-md-3 col-sm-3 text-center')[2]
+    capacity = ''
+    if capacity_html:
+        capacity = capacity_html.text.split(' ')[0].strip()
+
+    # extract the different rooms and their capacity in this venue
+    rooms_html = indiv_venue.find_all('ul', class_='space')
+    rooms = []
+    if rooms_html and len(rooms_html) > 1:
+        for index, room in enumerate(rooms_html[1].find_all('li')):
+            if room:
+                tokens = room.text.split(' ')
+                room_name = tokens[3].strip()
+                room_cap = tokens[0].strip()
+                rooms.append({
+                    "room" : room_name,
```

Comment:
Actually i dont really understand the website. This part of the code is extracting the part over here in the image below right (under capacity)? Does this mean they have 3 rooms?

From what i see, it doesnt seem to be. (Even if it is, we can just ignore it since they never provide any link to those rooms) Maybe for this scraper, you can just remove this code on extracting the different rooms? (I might also remove this as an attribute from the VenueClass.py because this attr doesnt seem useful >< )


![image](https://user-images.githubusercontent.com/40648338/87855852-9bb1b180-c94d-11ea-87d3-c32d4fca6096.png)

Reply:
Okay noted. I have removed the relevant code and left the roomName field empty - you can edit the VenueClass.py without worrying it would affect this VenuExplorer scraper!